### DISCUSSION TITLE
How to use "group-hover" when composing utility classes with @apply ?

### DISCUSSION THREAD

**Discussion by HanoonIH** [tailwindlabs/tailwindcss#2065] [Help]
I'm creating a card component by composing many utility classes using `@apply`.
I understand that I can make a component hover/active/focus effects using `.card:hover` , `.card:active` , `.card:focus` this way.
but how can I give a component, `group `and `group-hover` utilities...?
I tried `.card:group-hover` , but it isn't working.
Pls, help me.
@adamwathan @jackmcdade @r00k @philippbosch @davidhemphill

**Comment by philippbosch** [comment]
It would help if you gave more details about what you’re trying to achieve. Generally I would say `.group`/`.group-hover` are not needed in components but maybe I‘m missing something. 

Without components/`@apply` you would do something like this to e.g. highlight one child element when the user hovers over the card:

```html
<div class="bg-white border rounded group">
  <div class="font-bold text-teal-800 group-hover:text-red-500">Adam Wathan</div>
  <div class="text-teal-500">Tailwind Labs</div>
  <div class="text-gray-500">CEO</div>
</div>  
```

With a component, you would have something like this:

```html
<div class="card">
  <div class="card__name">Adam Wathan</div>
  <div class="card__company">Tailwind Labs</div>
  <div class="card__title">CEO</div>
</div>
```

and could achieve the same effect in your CSS:

```css
.card {
    @apply bg-white border rounded;
}

// …

.card:hover .card__name {
    @apply text-red-500;
}
```

Hope this helps.

**Reply by philippbosch** [reply]
@HanoonIH please either mark this as answered or provide more details
